binding/nodejs/index.js

```diff
@@ -0,0 +1,218 @@
+//
+// Copyright 2022 Picovoice Inc.
+//
+// You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+// file accompanying this source.
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+//
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+
+const PV_STATUS_T = require("./pv_status_t");
+const {
+  PvArgumentError,
+  PvStateError,
+  pvStatusToException,
+} = require("./errors");
+
+const { getSystemLibraryPath } = require("./platforms");
+
+const MODEL_PATH_DEFAULT = "lib/common/leopard_params.pv";
+
+const VALID_AUDIO_EXTENSIONS = [
+  ".flac",
+  ".mp3",
+  ".ogg",
+  ".opus",
+  ".vorbis",
+  ".wav",
+  ".webm"
+];
+
+/**
+ * Wraps the Leopard engine.
+ *
+ * Performs the calls to the Leopard node library. Does some basic parameter validation to prevent
+ * errors occurring in the library layer. Provides clearer error messages in native JavaScript.
+ */
+class Leopard {
+  /**
+   * Creates an instance of Leopard.
+   * @param {string} accessKey AccessKey obtained from Picovoice Console (https://console.picovoice.ai/).
+   * @param {string} manualModelPath the path to the Leopard model (.pv extension)
+   * @param {string} manualLibraryPath the path to the Leopard dynamic library (platform-dependent extension)
+   */
+  constructor(accessKey, manualModelPath, manualLibraryPath) {
+    if (
+      accessKey === null ||
+      accessKey === undefined ||
+      accessKey.length === 0
```

Comment:
what is `.length` is not defined?

Reply:
then `access_key` has an invalid type, and it goes all the way to napi, but then C spits out "invalid access_key" type. However, I updated `constructor` and `process` functions to check the types of input arguments to make sure that they're in a valid format.